contracts/schemes/SchemeFactory.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity ^0.5.16;
+
+import "@daostack/infra-experimental/contracts/votingMachines/IntVoteInterface.sol";
+import "@daostack/infra-experimental/contracts/votingMachines/VotingMachineCallbacksInterface.sol";
+import "../votingMachines/VotingMachineCallbacks.sol";
+import "../utils/DAOFactory.sol";
+import "@openzeppelin/upgrades/contracts/Initializable.sol";
+
+
+/**
+ * @title A factory and registrar for Schemes for organizations
+ * @dev The SchemeFactory is used for deploying and registering schemes at organizations
+ */
+
+contract SchemeFactory is Initializable, VotingMachineCallbacks, ProposalExecuteInterface {
+    event NewSchemeProposal(
+        address indexed _avatar,
+        bytes32 indexed _proposalId,
+        address indexed _intVoteInterface,
+        string _schemeName,
+        bytes _schemeData,
+        uint64[3] _packageVersion,
+        bytes4 _permissions,
+        address _schemeToReplace,
+        string _descriptionHash
+    );
+
+    event ProposalExecuted(address indexed _avatar, bytes32 indexed _proposalId, int256 _param);
+    event ProposalDeleted(address indexed _avatar, bytes32 indexed _proposalId);
+
+    // a SchemeProposal is a  proposal to add or remove a scheme to/from the an organization
+    struct SchemeProposal {
+        string schemeName;
+        bytes schemeData;
+        uint64[3] packageVersion;
+        address schemeToReplace;
+        bytes4 permissions;
+    }
+
+    mapping(bytes32=>SchemeProposal) public organizationProposals;
+
+    IntVoteInterface public votingMachine;
+    bytes32 public voteParams;
```

Comment:
so the params to remove are the same as param to register ?

Reply:
As we want to do everything in a single proposal, we can only have one params. We can't have one proposal voted on with 2 sets of params